provider/aws/aws.go

```diff
@@ -333,22 +355,41 @@ func (p *AWSProvider) Zones(ctx context.Context) (map[string]*route53.HostedZone
 				}
 			}
 
-			zones[aws.StringValue(zone.Id)] = zone
+			zones[aws.StringValue(zone.Id)] = &profiledZone{
+				profile: profile,
+				zone:    zone,
+			}
 		}
 
 		return true
 	}
 
-	err := p.client.ListHostedZonesPagesWithContext(ctx, &route53.ListHostedZonesInput{}, f)
-	if err != nil {
-		return nil, provider.NewSoftError(fmt.Errorf("failed to list hosted zones: %w", err))
-	}
-	if tagErr != nil {
-		return nil, provider.NewSoftError(fmt.Errorf("failed to list zones tags: %w", tagErr))
+	for p, client := range p.clients {
+		profile = p
+		err := client.ListHostedZonesPagesWithContext(ctx, &route53.ListHostedZonesInput{}, f)
+		if err != nil {
+			var awsErr awserr.Error
+			if errors.As(err, &awsErr) {
+				switch awsErr.Code() {
+				case "AccessDenied":
+					log.Warnf("Skipping AWS profile %q due to missing permission: %v", profile, awsErr.Message())
+					continue
+				case "InvalidClientTokenId", "ExpiredToken", "SignatureDoesNotMatch":
+					log.Warnf("Skipping AWS profile %q due to credential issues: %v", profile, awsErr.Message())
+					continue
+				default:
+					// noting to do here. Falling through to general error handling
```

Comment:
so you will wrap and ignore rest of the errors by SoftError, right?

Reply:
I would not say ignoring them, but yes, they're handled by `NewSoftError(...)`